internal/connector/connector.go

```diff
@@ -46,16 +45,20 @@ type Options struct {
 	SkipTLSVerify bool
 }
 
-type jwkCache struct {
+type authenticator struct {
 	mu          sync.Mutex
 	key         *jose.JSONWebKey
 	lastChecked time.Time
 
-	client  *http.Client
+	client  httpClient
```

Comment:
I made this an interface so that we can test it with a fake client implementation. In the future we could test it with an in-process server, but this was already an improvement in test coverage, so I left it for now.